### DISCUSSION TITLE
Start TTS generation as soon as 1 sentence worth of tokens are generated.

### DISCUSSION THREAD

**Discussion by chrishart0** [open-webui/open-webui#3345] [General]
**Is your feature request related to a problem? Please describe.**
The new audio generation features are awesome, but audio could generation much closer to real time. As it stands, text does not get sent to the audio generation API until the full response has been received. Preventing anything like a real time chat experience. 

**Describe the solution you'd like**
Since the TTS functionality already sends 1 sentence at a time to get generated, we should send the first sentence out to the audio API as soon as it has streamed in. That way, we can get audio back withing seconds of the message being sent instead of waiting for the whole response to come back. 

**Describe alternatives you've considered**
None

**Additional context**

**Technical Notes**

There are two places audio is generated which will need to be touched. The regular chat interface and the call interface. We will need to figure out how to tap into stream and, I suppose use the same sentence chunker which is already written chunking text to send to the TTS API for chunking the stream. I've only glanced at the code but this doesn't sound too hard. 

* https://github.com/open-webui/open-webui/blob/dev/src/lib/components/chat/Messages/ResponseMessage.svelte
* https://github.com/open-webui/open-webui/blob/dev/src/lib/components/chat/MessageInput/CallOverlay.svelte#L432

**Comment by thiswillbeyourgithub** [comment]
I fully support this request. Currently the TTS is barely usable for Claude Sonnet as it's so damn verbose.

**Comment by thiswillbeyourgithub** [comment]
No update on this? It doesn't look that complicated and seems like a huge improvement for interactivity (call mode notably)

**Reply by tjbck** [reply]
WebUI already has this.

**Reply by thiswillbeyourgithub** [reply]
Oups sorry and thank you

**Reply by thiswillbeyourgithub** [reply]
Actually @tjbck I checked and I'm not sure it's already the case. It takes about 3s when I click on "read aloud" for piper to output. But when I turn on the "auto playback response" and ask the LLM to tell me a story. It generates tokens for about 5s but I hear the audio only 8s after the start of the generation.

So I conclude that the tts calls still happens only at the end of the LLM generation. Right?

**Reply by tjbck** [reply]
"Auto playback response" only gets invoked once the completion has been entirely finished. If you'd like a faster response you should use the call mode.

**Reply by thiswillbeyourgithub** [reply]
Interesting. Thanks for the explanation

**Reply by gradshessians** [reply]
It would be nice if this were implemented not just for call mode as currently call mode suffers from picking up the audio being outputted by the speakers (maybe not an issue with a headset), resulting in a feedback loop where the llm's response becomes the next input.

**Reply by iChristGit** [reply]
> "Auto playback response" only gets invoked once the completion has been entirely finished. If you'd like a faster response you should use the call mode.

tested on 0.8.1, once the voice call emoji feature is enabled, it wont work and wait until end of LLM response, if no emoji feature yeah it quick!!
